### PR TITLE
fix: handle foreign key toward non primary key

### PR DESCRIPTION
**Pull request by GuillaumeCisco**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Create automatic tests
- [x] No automatic tests failures
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

File: services/analyzer/sequelize-tables-analyzer.js [hunk 1]
```diff
@@ -199,7 +202,7 @@ function createBelongsToReference(referenceTable, tableReferences, constraint) {
   const referenceColumnName = constraint.foreignColumnName;
   const referencePrimaryKeys = referenceTable.primaryKeys;
   const referenceUniqueConstraint = referenceTable.constraints
-    .find(({ columnType }) => columnType === 'UNIQUE');
+    .find(({ columnType }) => ['UNIQUE', 'PRIMARY KEY'].includes(columnType));
```
Comment (GuillaumeCisco) [inline services/analyzer/sequelize-tables-analyzer.js:205]: This is maybe not what we should do, I need confirmation on that

Comment (GuillaumeCisco) [reply services/analyzer/sequelize-tables-analyzer.js:205]: It looks fine, after some researches :)

File: test/services/dumper/dumper-sequelize.test.js [hunk 2]
```diff
@@ -53,6 +55,22 @@ describe('services > dumper > sequelize', () => {
     cleanOutput();
   });
 
+  it('should generate a model file with belongsTo associations and sourceKey/targetKey', async () => {
+    expect.assertions(2);
+    const dumper = getDumper();
+    await dumper.dump(simpleModelNonPrimary);
+    await dumper.dump(belongsToModelNonPrimary);
```
Comment (arnaud-moncel) [inline test/services/dumper/dumper-sequelize.test.js:62]: You can avoid the double usage of `dump` function by
```suggestion
    await dumper.dump({ ...simpleModelNonPrimary, ...belongsToModelNonPrimary });
```

**Review by arnaud-moncel — COMMENTED** [review]
Just a little comment, but it's look cool 😎 .
Feel free to take care of my suggestion.

File: test-expected/sequelize/dumper-output/owners.expected.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const { Sequelize } = sequelize;
+  // This section contains the fields of your model, mapped to your table's columns.
+  // Learn more here: https://docs.forestadmin.com/documentation/v/v6/reference-guide/models/enrich-your-models#declaring-a-new-field-in-a-model
+  const Owners = sequelize.define('owners', {
+    name: {
+      type: DataTypes.STRING,
+    },
+    ownerId: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
```
Comment (GuillaumeCisco) [inline test-expected/sequelize/dumper-output/owners.expected.js:13]: I wonder why dumper does not add a `unique: True` here

File: services/analyzer/sequelize-tables-analyzer.js (hunk 1 above)
Comment (GuillaumeCisco) [reply services/analyzer/sequelize-tables-analyzer.js:205]: 👍

File: test-expected/sequelize/dumper-output/owners.expected.js (hunk 3 above)
Comment (GuillaumeCisco) [reply test-expected/sequelize/dumper-output/owners.expected.js:13]: We do not need it when reading from a database ✅

**Review by arnaud-moncel — APPROVED** [review]
Let's go if tests are ok.

**Pull request merged**